services/sources/publisher_backlist_service.py

```diff
@@ -0,0 +1,110 @@
+from datetime import datetime, timedelta, timezone
+import os
+import requests
+import json
+import urllib.parse
+from typing import Optional, Dict
+
+from logger import create_log
+from mappings.UofM import UofMMapping
+from .source_service import SourceService
+
+
+logger = create_log(__name__)
+
+BASE_URL = "https://api.airtable.com/v0/appBoLf4lMofecGPU/Publisher%20Backlists%20%26%20Collections%20%F0%9F%93%96?view=All%20Lists"
+
+class PublisherBacklistService(SourceService):
+    def __init__(self):
+        self.airtable_auth_token = os.environ.get('AIRTABLE_KEY', None)
+
+    def get_records(
+        self,
+        full_import: bool=False, 
+        start_timestamp: datetime=None,
+        offset: Optional[int]=None,
+        limit: Optional[int]=None
+    ) -> list[UofMMapping]:
+        
+        array_json_records = self.get_records_json(full_import, start_timestamp, offset, limit)
+
+
+        for json_dict in array_json_records:
+            for records_value in json_dict['records']:
+                try:
+                    record_metadata_dict = records_value
+                    record = UofMMapping(record_metadata_dict)
+                    record.applyMapping()
+                except Exception as e:
+                    logger.exception(f'Failed to process Publisher Backlist record due to {e}')
```

Comment:
you can remove the due to {e} from this exception log because exception will also log the error message by default.